### ISSUE TITLE
No support for ALPN

### ISSUE THREAD

**Issue by timmattison** [eclipse-paho/paho.mqtt.java#858]
Paho doesn't currently allow the use of ALPN. This can be fixed in the SSLNetworkModule class. I will submit a PR for it.

- [x] Bug exists Release Version 1.2.5 ( Master Branch)
- [x] Bug exists in MQTTv3 Client on Snapshot Version 1.2.6-SNAPSHOT (Develop Branch)
- [x] Bug exists in MQTTv5 Client on Snapshot Version 1.2.6-SNAPSHOT (Develop Branch)

**Comment by timmattison** [comment]
This is necessary to connect to AWS IoT on port 443.

**Comment by timmattison** [comment]
On further review this appears to be quite involved to get working. The PR may take longer than expected. Has this been requested before?

**Comment by fpagliughi** [comment]
This was recently added to the Paho C library (and then the C++ and Rust libraries got it for "free"). But I haven't seen a request for the Java library yet.

**Comment by timmattison** [comment]
Yeah, this would be very helpful to connect to AWS IoT. Unfortunately it looks like it's beyond the level of effort I can expend for the time I was going to commit to it.

Is there any way to get this prioritized so Java has feature parity with C/C++/Rust?

**Comment by grandfield** [comment]
@timmattison Instead of adding ALPN support, do you think it would be appropriate to fix what caused it to break in the first place? See pull request https://github.com/eclipse/paho.mqtt.java/pull/942
I'm a newbie to paho, but something along these lines would be lower maintenance for PAHO. You are free to then add the aws alpn into the socket factory.
